e3sm_diags/driver/utils/dataset_xr.py

```diff
@@ -386,10 +386,10 @@ def _get_climo_dataset(self, season: str) -> xr.Dataset:
         filepath = self._get_climo_filepath(season)
         ds = xc.open_dataset(filepath, use_cftime=True, add_bounds=["X", "Y"])
 
-        if self.var in ds.variables:
-            pass
-        elif self.var in self.derived_vars_map:
+        if self.var in self.derived_vars_map:
             ds = self._get_dataset_with_derived_climo_var(ds)
+        elif self.var in ds.variables:
```

Comment:
Revert back the logic of derived variable as codes on main. This fix a problem that variables are not converted, for situations when self.var in ds.variables, but they still need units conversion or renaming.